apps/condo/domains/ticket/components/TicketClassifierSelect.tsx

```diff
@@ -115,28 +129,64 @@ const useTicketClassifierSelectHook = ({
     }
 }
 
+
+
 const CLASSIFIER_ROW_GUTTER: [Gutter, Gutter] = [40, 10]
 
 export const useTicketThreeLevelsClassifierHook = ({ initialValues: {
     classifierRule,
     placeClassifier,
     categoryClassifier,
     problemClassifier,
+    details,
 } }: ITicketThreeLevelsClassifierHookInput): ITicketThreeLevelsClassifierHookOutput => {
     const intl = useIntl()
     const PlaceClassifierLabel = intl.formatMessage({ id: 'component.ticketclassifier.PlaceLabel' })
     const CategoryClassifierLabel = intl.formatMessage({ id: 'component.ticketclassifier.CategoryLabel' })
     const ProblemClassifierLabel = intl.formatMessage({ id: 'component.ticketclassifier.ProblemLabel' })
-    const ruleRef = useRef({ id: classifierRule, place: null, category:null, problem: null })
+    const ruleRef = useRef<ITicketClassifierRuleType>({ id: classifierRule, place: null, category:null, problem: null })
     const client = useApolloClient()
     const ClassifierLoader = new ClassifiersQueryLocal(client)
     const validations = useTicketValidations()
     const ticketForm = useRef(null)
+    const hasUserSetClassifier = useRef<boolean>(false)
+    const hasPredictionFeature = hasFeature('classifier')
+
+    const stopPredict = useCallback(() => {
+        hasUserSetClassifier.current = true
+    }, [hasUserSetClassifier])
+
+    const predictTicketClassifier = async (details) => {
+        if (!hasPredictionFeature || !details || details.length < MIN_DESCRIPTION_LENGTH) {
+            return
+        }
+        if (ruleRef.current && (ruleRef.current.category || ruleRef.current.place) && hasUserSetClassifier.current) {
+            return
+        }
+        let prediction
+        try {
+            prediction = await client.query({
```

Comment:
Probably it’s better to move it to clientUtils

Reply:
Added to BackLog DOMA-2602 = as we need to move all direct usage of apollo client to client utils (also from classifier)